### PR TITLE
Banned role and copy button

### PR DESCRIPTION
**Pull request by orronai**

### CODE REVIEW

File: lms/lmsweb/views.py [hunk 1]
```diff
@@ -130,6 +130,15 @@ def favicon():
     )
 
 
+@webapp.before_request
+def banned_page():
+    if (
+        request.endpoint not in ('login', 'logout')
```
Comment (yammesicka) [inline lms/lmsweb/views.py:136]: I think checking if `current_user.is_authenticated and current.user.role.is_banned` might be more accurate

File: lms/templates/banned.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{% block page_content %}
+<div class="container">
+    <div id="login-container">
+        <div id="login" class="text-center">
+            <div id="course-logo" style="opacity: 0.6">
+                <img id="login-logo" src="{{ url_for('static', filename='avatar.jpg') }}" alt="תמונת הפרופיל של קורס פייתון" width="144" height="144">
+            </div>
+            <h1 id="main-title" class="h3 font-weight-normal">השהייה</h1>
+            <p>
+                המשתמש שלך מושהה עקב עבירה על תנאי הקורס.<br>
```
Comment (yammesicka) [inline lms/templates/banned.html:12]:
```
המשתמש שלך הושעה על ידי מנהל המערכת.
לפרטים נוספים אנא פנה אל צוות הניהול.
```

Generally I like the idea with the mailto, but I don't want the system to be specific for our course.
One of my next tasks is to to language localization and to remove the logo of our course ^^"

File: lms/templates/navbar.html [hunk 3]
```diff
@@ -30,21 +31,24 @@
           </div>
         </div>
       </li>
+      {% endif -%}
+      {%- if not current_user.role.is_banned %}
```
Comment (yammesicka) [inline lms/templates/navbar.html:35]: Why each section separately? Is this change really necessary?

File: lms/templates/view.html [hunk 4]
```diff
@@ -39,7 +39,10 @@ <h1>תצוגת תרגיל {{ solution['exercise']['id'] }}: {{ solution['exercis
   <div id="view-body">
     <div class="code-view-container">
       <div id="code-view" data-id="{{ solution['id'] }}" data-file="{{ current_file.id }}" data-exercise="{{ solution['exercise']['id'] }}" data-role="{{ role }}">
-        <pre><code class="language-python line-numbers highlight">{{- current_file.code | trim(chars=' ') | e -}}</code></pre>
+        <pre><code id="python-user-code" class="language-python line-numbers highlight">{{- current_file.code | trim(chars=' ') | e -}}</code></pre>
```
Comment (yammesicka) [inline lms/templates/view.html:42]: Let's try to remove this

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/templates/view.html [hunk 5]
```diff
@@ -39,7 +39,10 @@ <h1>תצוגת תרגיל {{ solution['exercise']['id'] }}: {{ solution['exercis
...
+      </div>
+      <div id="copy-code">
+        <button class="fa fa-copy" id="copy-button" title="Copy code" onclick="copy_code()" style="background-color:white;font-size:20px;"></button>
```
Comment (yammesicka) [inline lms/templates/view.html:45]: The style should be in the CSS, and the onclick should be in the JS file.

We try not to use inline CSS/JS.

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/lmsweb/views.py [hunk 6]
```diff
@@ -130,6 +130,16 @@ def favicon():
     )
 
 
+@webapp.before_request
+def banned_page():
+    if (
+        request.path != '/login'
```
Comment (yammesicka) [inline lms/lmsweb/views.py:136]: Is it really needed? We need to minimize the checks in `@before_request`, it is expensive.

File: lms/static/my.css [hunk 7]
```diff
@@ -344,6 +345,29 @@ i.fa {
   overflow-x: hidden;
 }
 
+.button-copy {
+  position: absolute;
+  right: 45px;
+  top: 8px;
+	background-color:gray;
```
Comment (yammesicka) [inline lms/static/my.css:352]: Fix the indentation

File: lms/static/my.js [hunk 8]
```diff
@@ -16,6 +16,19 @@ function escapeUnicode(str) {
 }
 
 
+function copyCode(button) {
+  button.addEventListener('click', () => {
+    var copyText = document.getElementById('python-user-code');
```
Comment (yammesicka) [inline lms/static/my.js:21]: Use `let` or `const`, not `var`. Also, if possible, define the variables as soon as possible.

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/templates/navbar.html [hunk 9]
```diff
@@ -1,4 +1,4 @@
-{% if request.path != '/login' %}
+{% if request.path != '/login' and not current_user.role.is_banned %}
```
Comment (yammesicka) [inline lms/templates/navbar.html:1]: If we change it already, change it to current_user.is_authenticated

File: lms/static/my.js [hunk 10]
```diff
@@ -59,4 +72,5 @@ window.escapeUnicode = escapeUnicode;
 window.addEventListener('load', () => {
   updateNotificationsBadge();
   trackReadAllNotificationsButton(document.getElementById('read-notifications'));
+  copyCode(document.getElementById('copy-button'));
```
Comment (yammesicka) [inline lms/static/my.js:75]: Follow the convention: change the name of the function to `trackCopyCodeButton`

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/templates/navbar.html [hunk 11]
```diff
@@ -1,4 +1,4 @@
-{% if request.path != '/login' %}
+{% if current_user.is_authenticated %}
```
Comment (yammesicka) [inline lms/templates/navbar.html:1]: and is not banned..

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/templates/navbar.html (hunk 11 above)
Comment (orronai) [reply lms/templates/navbar.html]: My fault...

**Review by yammesicka — APPROVED** [review]
Cool, good job :)

**Pull request merged**